internal/workspace_test.go

```diff
@@ -74,18 +74,92 @@ func TestAccWorkspaceIntegration(t *testing.T) {
 	assert.Contains(t, paths, notebook)
 }
 
+func TestAccWorkspaceUploadNotebookWithFileExtensionNoTranspile(t *testing.T) {
+	// TODO: remove NoTranspile suffix once other languages get Upload/Donwload features
+	ctx, w := workspaceTest(t)
+
+	notebookPath := filepath.Join("/Users", me(t, w).UserName, RandomName("notebook-")+".py")
+
+	err := w.Workspace.Upload(ctx, notebookPath, strings.NewReader("print(1)"))
+	assert.NoError(t, err)
```

Comment:
Does this add the header magically? If so this API won't work for regular files.

Reply:
https://github.com/databricks/databricks-sdk-go/blob/upload-download/service/workspace/utilities.go#L171-L175

Reply:
I mean the language header `# Databricks notebook source`.

Reply:
it's the backend that adds it

Reply:
Thanks -- now see that specifying the format does it, or having the file content header + format auto.